docs/UsersGuide/source/BuildSRW.rst

```diff
@@ -0,0 +1,489 @@
+   +--------------------------------+-----------------------------------------------------------+
+
+.. COMMENT: Is environment.yml deprecated? Remove?
+   Add rename_model.sh details. 
+
+.. _CheckoutExternals:
+
+Check Out External Components
+================================
+
+The SRW App relies on a variety of components (e.g., UFS_UTILS, ufs-weather-model, and UPP) detailed in :numref:`Chapter %s <Components>` of this User's Guide. Each component has its own repository. Users must run the ``checkout_externals`` script to collect the individual components of the SRW App from their respective GitHub repositories. The ``checkout_externals`` script uses the configuration file ``Externals.cfg`` in the top level directory of the SRW App to clone the correct tags (code versions) of the external repositories listed in :numref:`Section %s <HierarchicalRepoStr>` into the appropriate directories (e.g., ``ush``, ``sorc``). 
+
+Run the executable that pulls in SRW App components from external repositories:
+
+.. code-block:: console
+
+   cd $SRW
+   ./manage_externals/checkout_externals
+
+The script should output dialogue indicating that it is retrieving different code repositories. It may take several minutes to download these repositories.
+
+To see more options for the ``checkout_externals`` script, users can run ``./manage_externals/checkout_externals -h``. For example:
+
+   * ``-S``: Outputs the status of the repositories managed by ``checkout_externals``. By default only summary information is provided. Use with the ``-v`` (verbose) option to see details.
+   * ``-x [EXCLUDE [EXCLUDE ...]]``: allows users to exclude components when checking out externals. 
+   * ``-o``: By default only the required externals are checked out. This flag will also checkout the optional externals.
+
+Generally, users will not need to use the options and can simply run the script, but the options are available for those who are curious. 
+
+.. _BuildExecutables:
+
+Set Up the Environment and Build the Executables
+===================================================
+
+.. _DevBuild:
+
+``devbuild.sh`` Approach
+-----------------------------
+
+On Level 1 systems for which a modulefile is provided under the ``modulefiles`` directory, users can build the SRW App binaries with the following command:
+
+.. code-block:: console
+
+   ./devbuild.sh --platform=<machine_name>
+
+where ``<machine_name>`` is replaced with the name of the platform the user is working on. Valid values are: ``cheyenne`` | ``gaea`` | ``hera`` | ``jet`` | ``linux`` | ``macos`` | ``noaacloud`` | ``odin`` | ``orion`` | ``singularity`` | ``wcoss2``
+
+.. note::
+   Although build modulefiles exist for generic Linux and MacOS machines, users will need to alter these according to the instructions in Sections :numref:`%s <CMakeApproach>` & :numref:`%s <MacDetails>`. It is recommended that users on these systems build the SRW App with the :ref:`CMake Approach <CMakeApproach>` instead. 
```

Comment:
Yeah, I think it would be easier to have just one approach. @natalie-perlin do you think there's any reason to keep the CMake approach? It seems like even on generic Mac/Linux systems, users can just adapt the build files and use the devbuild approach.

Reply:
@danielabdi-noaa @gspetro-NOAA - 
devbuild.sh is indeed the fastest to build the SRW when everything is known to build properly on a given system and setup. 
For all other cases, there are two reasons to keep the CMake approach:
1) verification that Lmod is installed and functional, when you 
`source ./etc/lmod-setup.sh <machine>,` or otherwise initialialize Lmod
2) verification that all the software stack modules are found and load properly. 
```
module use <SRW>/modulefiles
module load build_<machine>_<compiler>
```
The latter is especially important after transition to use of Lua modules from earlier TCL format. With the Lua format, when if any module is not found among those listed in `build_<system>_<compiler>.lua` , then none of the modules is loaded. 

Use of CMake approach allows to separate steps when building the SRW, for debugging purposes .

Reply:
I'd like to offer an alternative opinion.  There is no such thing as phasing out the direct CMake build approach because that is how the App is built, whether the user is typing the commands or the `debvbuild` script is running them.  I view the `devbuild` as a convenience script for a very specific set of use cases and nothing more.  It can be good for providing a shortcut for a selected group of machines, but it is also borderline over-engineering, and has a lot of issues in portability, scalability, and maintenance (keeping up with growing list of platforms and keeping up to date with changes to those supported platforms).  It is also completely useless when taking the UFS to a build environment that differs from the ones captured in the convenience script.  There are many reasons why a researcher may want to run the UFS on a new machine, or even a "supported" machine using a different set of modules for research purposes.  The `devbuild` script is really only useful for a very specific set of use cases that are restricted to running/developing for very basic types of situations where users are just running the App with the very limited set of platforms/software provided.  Users should not have to figure out how to modify code (the `devbuild` script) to build the system on a new platform or with a different software stack.  It's already confusing for some researchers to see an opaque layer on top of the industry standard CMake build mechanism and hiding the means to directly access the very simple standard build that everyone expects is not helping users, it's hindering them.  

If I had to pick one, I'd remove the `devbuild` script because it adds a small amount of convenience at the expense of unnecessary portability, maintenance, and complexity problems for developers, and obfuscation problems for users wanting to do something `devbuild` doesn't offer.

Reply:
In general, I like the cleanness of offering only one option, and when devbuild works, it is so much easier (at least for me--and I've used both as a relatively unsophisticated user). However, I do think that in this chapter, we're erring on the side of "more information is better," and the CMake option does provide more transparency with what is happening under the hood, which is important if the app is not building or if users are trying to make it work on a platform that we have not tested (like a generic Linux or Mac that we haven't tried to build/run on). For the quick/clean/straightforward approach, users can look at the Quickstart chapter. For now, I'll keep both options barring a mandate from @MichaelLueken to the contrary. 
I think we'd eventually like to make a troubleshooting chapter, at which point it might make sense to move the CMake stuff over there (TBD). But for now it can't hurt to have both, especially since the RTD version has clear notes telling users whether they need to view a section and hyperlinks for where to go next if they don't (hard to see this on GitHub though).